utils/error_injection.hh

```diff
@@ -104,6 +105,14 @@ using error_injection_parameters = std::unordered_map<sstring, sstring>;
  *    Requires func to be a function taking an injection_handler reference and
  *    returning a future<>.
  *    Expected use case: wait for an event from tests.
+ * 5. inject_parameter(name)
```

Comment:
heads up: in recently merged https://github.com/scylladb/scylladb/pull/17734, `inject_with_handler` was renamed into `inject`, maybe we want to go in this direction here too? I have no opinion

Reply:
(it's currently on the `next` queue, there could also be a conflict)

Reply:
I don't like when different things are called the same. `inject_with_handler` was behaving quite differently from other `inject`-s. I certainly want a separate name for this new function.